Ctor 457 new pack network juniper mseries netconf (#4416)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/network-routers-juniper-mseries-snmp.md

> import Tabs from '@theme/Tabs';

Reviewer: Juniper M-Series SNMP?

Reviewer 2: SAAS-1970

import TabItem from '@theme/TabItem';

## Connector dependencies

The following monitoring connectors will be installed when you install the **Juniper M-Series** connector through the

> **Configuration > Monitoring Connectors** menu:

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

> **Configuration > Monitoring Connectors** menu:

Reviewer 2: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

| Service Alias  | Service Template                          | Service Description                  |
|:---------------|:------------------------------------------|:-------------------------------------|

> | Cpu-Routing    | Net-Juniper-Mseries-Cpu-Routing-custom    | Check CPU Usage of routing engine    |

Reviewer: ```suggestion
| Cpu-Routing    | Net-Juniper-Mseries-Cpu-Routing-custom    | Check CPU usage of routing engine    |
```

> | Cpu-Routing    | Net-Juniper-Mseries-Cpu-Routing-custom    | Check CPU Usage of routing engine    |

Reviewer 2: ```suggestion
| Cpu-Routing    | Net-Juniper-Mseries-Cpu-Routing-custom    | Check CPU usage of routing engine    |
```

| Hardware       | Net-Juniper-Mseries-Hardware-custom       | Check hardware                       |

> | Memory-Routing | Net-Juniper-Mseries-Memory-Routing-custom | Check Memory Usage of routing engine |

Reviewer: ```suggestion
| Memory-Routing | Net-Juniper-Mseries-Memory-Routing-custom | Check memory usage of routing engine |
```

> | Memory-Routing | Net-Juniper-Mseries-Memory-Routing-custom | Check Memory Usage of routing engine |

Reviewer 2: ```suggestion
| Memory-Routing | Net-Juniper-Mseries-Memory-Routing-custom | Check memory usage of routing engine |
```

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias              | Service Template                                           | Service Description                                                                                                                       | Discovery  |
|:---------------------------|:-----------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------|:----------:|
| Bgp-Peer-Prefix-Statistics | Net-Juniper-Mseries-SNMP-Bgp-Peer-Prefix-Statistics-custom | Check BGP peers prefix statistics                                                                                                         |            |
| Bgp-Peer-State             | Net-Juniper-Mseries-SNMP-Bgp-Peer-State-custom             | Check BGP peers status                                                                                                                    |            |

> | Disk-Generic-Id            | Net-Juniper-Mseries-Disk-Generic-Id-custom                 | Check the rate of free space on the disk (use the ID). Thresholds can be in percentage or in free space remaining                         |            |

Reviewer: ```suggestion
| Disk-Generic-Id            | Net-Juniper-Mseries-Disk-Generic-Id-custom                 | Check the rate of free space on the disk (use the ID). Thresholds can be in percentage or in free remaining space                        |            |
```

> | Disk-Generic-Id            | Net-Juniper-Mseries-Disk-Generic-Id-custom                 | Check the rate of free space on the disk (use the ID). Thresholds can be in percentage or in free space remaining                         |            |

Reviewer 2: ```suggestion
| Disk-Generic-Id            | Net-Juniper-Mseries-Disk-Generic-Id-custom                 | Check the rate of free space on the disk (use the ID). Thresholds can be in percentage or in free remaining space                        |            |
```

| Disk-Generic-Name          | Net-Juniper-Mseries-Disk-Generic-Name-custom               | Check the rate of free space on the disk (use the Name. Difficult to use                                                                  |            |

> | Disk-Global                | Net-Juniper-Mseries-Disk-Global-custom                     | Check the rate of free space on the disk (use the Name. Difficult to use                                                                  | X          |

Reviewer: difficult to use?

> | Disk-Global                | Net-Juniper-Mseries-Disk-Global-custom                     | Check the rate of free space on the disk (use the Name. Difficult to use                                                                  | X          |

Reviewer 2: ```suggestion
| Disk-Generic-Name          | Net-Juniper-Mseries-Disk-Generic-Name-custom               | Check the rate of free space on the disk (use the Name.                                                                   |            |
| Disk-Global                | Net-Juniper-Mseries-Disk-Global-custom                     | Check the rate of free space on the disk (use the Name.                                                              | X          |
```

</TabItem>
<TabItem value="Rsvp-Session-Status" label="Rsvp-Session-Status">

| Name   | Unit  |
|:-------|:------|
| status | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Traffic-*" label="Traffic-*">

> Coming soon

Reviewer 2: ```suggestion
| Name                                          | Unit  |
|:-----------------------------------------------------|:------|
| *interface_name*#status                              | N/A   |
| *interface_name*#interface.traffic.in.bitspersecond  | b/s   |
| *interface_name*#interface.traffic.out.bitspersecond | b/s   |

> Applies to the following service templates: Traffic-Generic-Id, Traffic-Generic-Name, Traffic-Global
```

### Pack

The installation procedures for monitoring connectors are slightly different depending on [whether your license is offline or online](../getting-started/how-to-guides/connectors-licenses.md).

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the

> **Configuration > Monitoring Connectors** menu.

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

> **Configuration > Monitoring Connectors** menu.

Reviewer 2: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

</Tabs>

2. Whatever the license type (*online* or *offline*), install the **Juniper M-Series** connector through

> the **Configuration > Monitoring Connectors** menu.

Reviewer: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

> the **Configuration > Monitoring Connectors** menu.

Reviewer 2: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

| FILTERLOCALAS  | Filter by local AS (can be regexp)                                                                                                                                                                                                                                                                                                                                                                             |                                                                     |             |
| CRITICALSTATUS | Specify critical threshold (default: C\<'%\{peer\_status\} =~ /running/ && %\{peer\_state\} !~ /established/'\>). Can use special variables like C\<%\{peer\_identifier\}\>, C\<%\{peer\_state\}\>, C\<%\{peer\_status\}\>, C\<%\{local\_type\}\>, C\<%\{local\_ip\}\>, C\<%\{local\_port\}\>, C\<%\{local\_as\}\>, C\<%\{remote\_type\}\>, C\<%\{remote\_ip\}\>, C\<%\{remote\_port\}\>, C\<%\{remote\_as\}\> | %\{peer\_status\} =~ /running/ && %\{peer\_state\} !~ /established/ |             |

> | WARNINGSTATUS  | Specify warning threshold. Can use special variables like C\<%\{peer\_identifier\}\>, C\<%\{peer\_state\}\>, C\<%\{peer\_status\}\>, C\<%\{local\_type\}\>, C\<%\{local\_ip\}\>, C\<%\{local\_port\}\>, C\<%\{local\_as\}\>, C\<%\{remote\_type\}\>, C\<%\{remote\_ip\}\>, C\<%\{remote\_port\}\>, C\<%\{remote\_as\}\>                                                                                        |                                                                     |             |

Reviewer: C?

> | WARNINGSTATUS  | Specify warning threshold. Can use special variables like C\<%\{peer\_identifier\}\>, C\<%\{peer\_state\}\>, C\<%\{peer\_status\}\>, C\<%\{local\_type\}\>, C\<%\{local\_ip\}\>, C\<%\{local\_port\}\>, C\<%\{local\_as\}\>, C\<%\{remote\_type\}\>, C\<%\{remote\_ip\}\>, C\<%\{remote\_port\}\>, C\<%\{remote\_as\}\>                                                                                        |                                                                     |             |

Reviewer 2: ```suggestion
 CRITICALSTATUS | Specify critical threshold (default: '%\{peer_status\} =~ /running/ && %\{peer_state\} !~ /established/'). Can use special variables like %\{peer_identifier\}, %\{peer_state\}, %\{peer_status\}, %\{local_type\}, %\{local_ip\}, %\{local_port\}, %\{local_as\}, %\{remote_type\}, %\{remote_ip\}, %\{remote_port\}, %\{remote_as\} | %\{peer_status\} =~ /running/ && %\{peer_state\} !~ /established/ |             |
| WARNINGSTATUS  | Specify warning threshold. Can use special variables like %\{peer_identifier\}, %\{peer_state\}, %\{peer_status\}, %\{local_type\}, %\{local_ip\}, %\{local_port\}, %\{local_as\}, %\{remote_type\}, %\{remote_ip\}, %\{remote_port\}, %\{remote_as\}                                                                               |                                                                 |             |
```

| CRITICAL     | Critical threshold                                                                                                                                                                              | 95                     |             |
| WARNING      | Warning threshold                                                                                                                                                                               | 90                     |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                                                                                              | --verbose              |             |

</TabItem>
<TabItem value="Hardware" label="Hardware">

| Macro        | Description                                                                                        | Default value     | Mandatory   |
|:-------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | COMPONENT    | Which component to check (default: C\<'.*'\>). Can be: `fru`, `operating`, `alarm`                 | .*                |             |

Reviewer: C?

> | COMPONENT    | Which component to check (default: C\<'.*'\>). Can be: `fru`, `operating`, `alarm`                 | .*                |             |

Reviewer 2: ```suggestion
| COMPONENT    | Which component to check. Can be: `fru`, `operating`, `alarm`                 | .*                |             |
```

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose         |             |

</TabItem>
<TabItem value="Ldp-Session-Status" label="Ldp-Session-Status">

| Macro              | Description                                                                                                                                             | Default value                | Mandatory   |
|:-------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------|:-----------:|
| FILTERENTITY       | Threshold                                                                                                                                               |                              |             |

> | FILTERPEER         | Threshold                                                                                                                                               |                              |             |

Reviewer 2: ```suggestion
| FILTERENTITY       | Filter entities                                                                                                                         |                            |             |
| FILTERPEER         | Filter peers                                                                                                                         |                            |             |
```

| FILTERFROM              | Threshold                                                                                                                                                |                     |             |
| FILTERTO                | Threshold                                                                                                                                                |                     |             |
| WARNINGLASTTRANSITION   | Threshold                                                                                                                                                |                     |             |
| CRITICALLASTTRANSITION  | Threshold                                                                                                                                                |                     |             |
| CRITICALSTATUS          | Define the conditions to match for the status to be CRITICAL (default: C\<'%\{state\} !~ /up/i'\>). You can use the following variables: C\<%\{state\}\> | %\{state\} !~ /up/i |             |

> | WARNINGSTATUS           | Define the conditions to match for the status to be WARNING (default: `''`). You can use the following variables: C\<%\{state\}\>                        |                     |             |

Reviewer 2: ```suggestion
| CRITICALSTATUS          | Define the conditions to match for the status to be CRITICAL (default: '%\{state\} !~ /up/i'). You can use the following variables: %\{state\} | %\{state\} !~ /up/i |             |
| WARNINGSTATUS           | Define the conditions to match for the status to be WARNING. You can use the following variables: %\{state\}                   |                   |             |
```

> | WARNINGSTATUS           | Define the conditions to match for the status to be WARNING (default: `''`). You can use the following variables: C\<%\{state\}\>                        |                     |             |

Reviewer: C?

| CRITICALSTATUS | Define the conditions to match for the status to be CRITICAL (default: '%\{state\} !~ /up/i'). You can use the following variables: %\{state\} | %\{state\} !~ /up/i |             |
| WARNINGSTATUS  | Define the conditions to match for the status to be WARNING (default: ''). You can use the following variables: %\{state\}                     |                     |             |
| EXTRAOPTIONS   | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                                             |                     |             |

</TabItem>
<TabItem value="Traffic-Generic-Id" label="Traffic-Generic-Id">

| Macro        | Description                                                                                        | Default value     | Mandatory   |
|:-------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | INTERFACEID  | Set the interface (number expected) example: 1,2,... (empty means 'check all interfaces')          |                   |             |

Reviewer: same as FR
